app/assets/javascripts/models/beta/iteration.js

```diff
@@ -182,13 +218,9 @@ export const groupBySprints = (stories = [], project, initialSprintNumber = 1, p
     const firstSprintIndex = 0;
     const isFromSprintInProgress = !Story.isUnstarted(story);
 
-    sprints = handleSprintsOverflow(
-      project,
-      sprints,
-      Story.getPoints(story),
-      initialSprintNumber,
-      velocity
-    );
+    if (hasOverflow(story, velocity) && !isFromSprintInProgress) {
+      return handleSprintsOverflow(project, sprints, story, velocity, initialSprintNumber);
```

Comment:
I think this name could be improved as well, "handle" is not a very descriptive name for what it does for real. What do you think about `addSprintWithOverflow`?

Reply:
Yes, is better, and will follow the pattern of other methods: `addStoryToSprint` `addToSprintFromBacklog`